chain/network/src/config.rs

```diff
@@ -168,6 +168,48 @@ pub struct NetworkConfig {
 }
 
 impl NetworkConfig {
+    /// Overrides values of NetworkConfig with values for the JSON config.
+    /// We need all the values from NetworkConfig to be configurable.
+    /// We need this in case of emergency. It is faster to change the config than to recompile.
+    fn override_config(mut self, overrides: crate::config_json::NetworkConfigOverrides) -> Self {
```

Comment:
[stylistic personal preference]
`fn override(mut self) -> Self {}` works, but I'd go for `fn override(&mut self) {}` to make the callsite less confusing.
```
x.override();
Ok(x)
```

seems clearer to me than
```
Ok(x.override())
```